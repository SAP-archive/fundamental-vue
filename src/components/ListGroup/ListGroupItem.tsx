import Vue from 'vue';
import { Slot, Component, DefaultSlot } from '@/core';

@Component('ListGroupItem')
@DefaultSlot('Content displayed by the item. Usually text.')
@Slot('action', 'Custom actions (displayed on the right side, usually a button)')
export class ListGroupItem extends Vue {
  public render() {
    const action = this.$slots.action;
    return (
      <li class='fd-list-group__item'>
        {this.$slots.default}
        {!!action && <span class='fd-list-group__action'>{action}</span>}
      </li>
    );
  }
}
