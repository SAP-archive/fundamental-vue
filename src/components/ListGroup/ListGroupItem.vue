<template>
  <li class="fd-list-group__item">
    <slot />
    <span v-if="hasAction" class="fd-list-group__action">
      <slot name="action" />
    </span>
  </li>
</template>

<script>
export default {
  name: "FdListGroupItem",
  computed: {
    hasAction() {
      return this.$slots.action != null;
    }
  }
};
</script>
