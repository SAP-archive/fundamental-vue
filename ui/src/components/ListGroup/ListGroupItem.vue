<template>
  <li class="fd-list-group__item">
    <slot />
    <span v-if="hasAction" class="fd-list-group__action">
      <slot name="action" />
    </span>
  </li>
</template>

<script lang="ts">
import Vue from "vue";
export default Vue.extend({
  name: "FdListGroupItem",
  computed: {
    hasAction(): boolean {
      return this.$slots.action != null;
    }
  }
});
</script>
