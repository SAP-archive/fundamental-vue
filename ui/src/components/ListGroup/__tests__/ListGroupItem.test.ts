import { mount } from "@vue/test-utils";
import ListGroupItem from "../ListGroupItem.vue";
import { CreateElement } from "vue";

describe("ListGroup", () => {
  const createActionElement = (h: CreateElement, index: number) => {
    return h(
      "div",
      {
        class: "action-element",
        slot: "action"
      },
      `Action ${index}`
    );
  };

  const wrapper = mount(ListGroupItem, {
    slots: {
      default: "List Item 1",
      action: {
        render(h: CreateElement) {
          const actionElement1 = createActionElement(h, 1);
          const actionElement2 = createActionElement(h, 2);
          return h("div", {}, [actionElement1, actionElement2]);
        }
      }
    }
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("has correct number of actions", () => {
    expect(wrapper.findAll(".action-element").length).toBe(2);
  });
});
