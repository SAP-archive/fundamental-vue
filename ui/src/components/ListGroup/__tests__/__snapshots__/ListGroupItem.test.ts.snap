// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroup renders correctly 1`] = `
<li
  class="fd-list-group__item"
>
  List Item 1 
  <span
    class="fd-list-group__action"
  >
    <div>
      <div
        class="action-element"
      >
        Action 1
      </div>
      <div
        class="action-element"
      >
        Action 2
      </div>
    </div>
  </span>
</li>
`;
